### PR TITLE
Alternative web ui for scraped data

### PR DESCRIPTION
**Pull request by scotthooker**
Alternative web ui for scraped data

### CODE REVIEW

**Comment by rmcdonald1442** [comment]
Wow this looks phenomenal! The infinite scrolling is flawless it seems.

image

**Comment by scotthooker** [comment]
Oh I think it’s actually cause I’m just loading 2500 results so the infinite scroll isn’t actually doing anything… all results are loaded at the start…

I maybe wrong though. Locally I changed it to load all can’t remember what I have in the PR

File: src/alternative-web/src/components/Pagination.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { usePropertyContext } from '../context/PropertyContext';
+
+const Pagination: React.FC = () => {
+  const { state, dispatch } = usePropertyContext();
+  const itemsPerPage = 2500;
```
Comment (scotthooker) [inline src/alternative-web/src/components/Pagination.tsx:6]: Ah yeh that be why… cheating a little there

**Comment by scotthooker** [comment]
There’s probably a few more issues with filtering and sorting but feel free to merge if you want to. Can run both uis side by side. I think we both agree it’s not a polished project so can just be used with flaws and bugs. Sold as seen you could say!

File: src/alternative-web/src/components/Pagination.tsx (hunk 1 above)
Comment (rmcdonald1442) [reply src/alternative-web/src/components/Pagination.tsx:6]: Ah that's fine haha. It's a big improvement of the existing pagination and at least all the images don't render at once.

File: src/alternative-web/src/context/PropertyContext.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    if (state.showFavourite && !item.state.isFavourite) return false;
+    if (!state.showHidden && item.state.isHidden) return false;
+    if (state.showMissingFtg && item.listing.squareFootage !== null) return false;
+    if (state.showMissingRoomDimensions && item.listing.rooms && item.listing.rooms.every((room: Room) => room.dimensions !== null)) return false;
+    if (state.minSquareFootage && item.listing.squareFootage && item.listing.squareFootage < parseFloat(state.minSquareFootage)) return false;
+    if (state.filterDate && item.listing.addedOrReduced && new Date(item.listing.addedOrReduced) < new Date(state.filterDate)) return false;
+    if (state.searchTerm && !item.listing.title.toLowerCase().includes(state.searchTerm.toLowerCase())) return false;
+
+    // Apply category filters
+    if (Object.values(state.categoryFilters).some(value => value)) {
```
Comment (rmcdonald1442) [inline src/alternative-web/src/context/PropertyContext.tsx:149]: Nice idea!

**Comment by rmcdonald1442** [comment]
@scotthooker awesome, I've merged in. I'll start making any future UI changes to this version.

If you need to extend any of the existing interfaces / types feel free to do so.

**Comment by scotthooker** [comment]
If you can trace the bug thats duplicating the listings. I think somewhere im adding dupe records into the context.

**Comment by scotthooker** [comment]
Also, DM me back on LinkedIn. See if I can grab you discord or soemthing for a more instant chat.

**Comment by rmcdonald1442** [comment]
@scotthooker that'd be great! I'm on holiday until the 30th and haven't been online much but I will message you after then with a discord link or something.

**Comment by rmcdonald1442** [comment]
> If you can trace the bug thats duplicating the listings. I think somewhere im adding dupe records into the context.

Hmm if you're talking about duplicates in the same website I think it might be here: 

`const importedOnTheMarketData = (await import('../../../../storage/datasets/all-onthemarket/000000001.json')).default.listings;`

We should instead use `current-onthemarket`. 

Basically during the data collection step I had in mind that that the `all-onthemarket/zoopla/rightmove` datasets were a sort of complete log, and at the end of a batch this log gets consolodated down into a `current-....` dataset.

By "consolidate" I mean removing duplicates, combining entries with incomplete info, updating the current dataset with the most up-to-date info (mostly price).

The reasoning behind this was I could maintain a history of "all" listings and a "current" state of the listings. For example every month I might look at every property again and see if there are any price changes, sold properties, etc.

But I haven't donen anything with this approach, so in the end it just looks messy, and inefficient.

**Pull request merged**